### DISCUSSION TITLE
TMDL Workspace Connection

### DISCUSSION THREAD

**Discussion by suketududhat** [TabularEditor/TabularEditor3#1002] [🙋♀️TE3 Q&A]
Hi, we use TE3 with workspace connection to PBI Premium for Datasets. I was testing TMDL save to folder but reopening the .tmd file in TE3 does not prompt me to choose a workspace database/dataset. Is this a planned feature or is this not possible with the TMDL file structure compared to a single .bim file?

**Comment by otykier** [comment]
Hi @suketududhat
This is a bug when you use **File > Open > Model from File...**.

As a workaround for now, use **File > Open > Model from Folder...**

<img width="388" alt="image" src="https://github.com/TabularEditor/TabularEditor3/assets/8976200/1872aaf7-def7-45f0-a39c-809e9604c4d9">

**Comment by suketududhat** [comment]
Hi @otykier, great workaround and it does get me further in the process but it is causing a new issue now in screenshot. 

![image](https://github.com/TabularEditor/TabularEditor3/assets/60863909/5bf2b922-4bee-4246-8a55-bc5edc725529)

I am trying to open a Power BI Dataset from a Premium workspace connection. I tried opening both an existing dataset and creating a new one but both give the same error.

**Reply by otykier** [reply]
This, I believe, is an issue with how TMDL serializes and deserializes explicit data sources. My understanding is that this is still not fully supported in the TMDL preview, but in any case, you should consider raising this as an issue with Microsoft: https://github.com/microsoft/analysis-services
